src/fw_update.c

```diff
@@ -122,6 +122,45 @@ static bool manifest_version_is_different(const struct golioth_ota_manifest *man
     return false;
 }
 
+static void fw_report_and_observe(void)
+{
+    enum golioth_status status;
+    int retry_count = 3;
+
+    status = golioth_fw_update_report_state_sync(_client,
+                                                 GOLIOTH_OTA_STATE_IDLE,
+                                                 GOLIOTH_OTA_REASON_READY,
+                                                 _config.fw_package_name,
+                                                 _config.current_version,
+                                                 NULL,
+                                                 GOLIOTH_SYS_WAIT_FOREVER);
+
+    if (status != GOLIOTH_OK)
+    {
+        GLTH_LOGE(TAG, "Failed to report firmware state: %d", status);
+    }
+
+    while (retry_count > 0)
+    {
+        status = golioth_ota_observe_manifest_async(_client, on_ota_manifest, NULL);
+
+        if (status == GOLIOTH_OK)
+        {
+            break;
+        }
+        else if (retry_count == 1)
+        {
+            GLTH_LOGE(TAG, "Failed to observe firmware manifest: %d", status);
+        }
+        else
+        {
+            GLTH_LOGW(TAG, "Failed to observe manifest, will retry. %d", status);
+            golioth_sys_msleep(2000);
```

Comment:
Let's make this delay longer. It's not so important if we observe immediately after a reboot, but is important that we do eventually observe. Maybe 5 seconds?